docs/sql-migration-guide-upgrade.md

```diff
@@ -126,6 +126,8 @@ license: |
 
   - Since Spark 3.0, parquet logical type `TIMESTAMP_MICROS` is used by default while saving `TIMESTAMP` columns. In Spark version 2.4 and earlier, `TIMESTAMP` columns are saved as `INT96` in parquet files. To set `INT96` to `spark.sql.parquet.outputTimestampType` restores the previous behavior.
 
+  - Since Spark 3.0, a new data source option `pathGlobFilter` is introduced for filtering files in `DataFrameReader` and `DataStreamReader`. For example, `spark.read.option("pathGlobFilter", "*.orc").orc(path)` will read all the files ending with `.orc` under the given `path`. Note that with the option the query result will contain partition columns if any; while with glob pattern in path, e.g `spark.read.orc("path/*/*/*/*.orc")`, the result won't contain partition columns.
```

Comment:
Was this a behaviour change?

Reply:
It is not. I think it is kind of useful for users.  I can remove this if you insist.

Reply:
Yes it's useful but here's migration guide. Let's just document them somewhere else.

Reply:
This does not introduce any behavior change. Thus, I would prefer to document it in the SQL guide: https://spark.apache.org/docs/latest/sql-data-sources-load-save-functions.html 

I think we can add a section for documenting the useful options for our built-in file sources.

Reply:
I will update  https://spark.apache.org/docs/latest/sql-data-sources-load-save-functions.html. Make this PR as WIP for now.